File: app/src/main/java/net/emilla/lang/measure/CelsiusConversion.java
package net.emilla.lang.measure;

public record CelsiusConversion(double degrees, boolean fromKelvin) {

    public double convert() {
        return fromKelvin
            ? degrees - 273.15
            : (degrees - 32.0) * 5.0 / 9.0;
    }

}

File: app/src/main/java/net/emilla/lang/measure/CelsiusConversion.kt
package net.emilla.lang.measure

data class CelsiusConversion(@JvmField val degrees: Double, @JvmField val fromKelvin: Boolean) {

    fun convert() = if (fromKelvin) degrees - 273.15 else (degrees - 32.0) * 5.0 / 9.0
}
